flink-yarn/src/main/java/org/apache/flink/yarn/YarnResourceManager.java

```diff
@@ -207,11 +210,14 @@ protected void shutDownApplication(ApplicationStatus finalStatus, String optiona
 	@Override
 	public void startNewWorker(ResourceProfile resourceProfile) {
 		// Priority for worker containers - priorities are intra-application
-		//TODO: set priority according to the resource allocated
 		Priority priority = Priority.newInstance(generatePriority(resourceProfile));
 		int mem = resourceProfile.getMemoryInMB() < 0 ? DEFAULT_TSK_EXECUTOR_MEMORY_SIZE : (int)resourceProfile.getMemoryInMB();
 		int vcore = resourceProfile.getCpuCores() < 1 ? 1 : (int)resourceProfile.getCpuCores();
 		Resource capability = Resource.newInstance(mem, vcore);
+
+		if (!orgPriorityResources.containsKey(priority.getPriority())) {
+			orgPriorityResources.put(priority.getPriority(), capability);
```

Comment:
Isn't `capability` basically the same as `resourceProfile`? Why storing the different type? `orgPriorityResources` is effectively the inverse map of `resourcePriorities`, right? Why not treating them as a proper `BiMap`?